### PR TITLE
enhancement(gcp_stackdriver_logs sink): Add severity key

### LINKED ISSUE #2678: bug(sinks gcp_stackdriver_logs): Cannot set log messages severity level
Hi Vector team,

When using the ```gcp_stackdriver_logs``` sink, all messages have a severity level of ```DEFAULT``` as the severity property of the ```LogEntry``` is not being set.

I can set a severity field, but it's added to the ```jsonPayload``` property of the ```LogEntry``` and stackdriver is not gettint that as the message severity.

**Proposed solution**

We should have a ```severity``` configuration with this posible values:

```
DEFAULT  # default configuration value
DEBUG
INFO
NOTICE
WARNING
ERROR
CRITICAL
ALERT
EMERGENCY
```

Thank you all

### PR DESCRIPTION
**Pull request by bruceg**
This adds support for a configurable `severity_key` field name, from
which the outgoing log severity will be taken.

Signed-off-by: Bruce Guenter <bruce@timber.io>

Closes #2678

### CODE REVIEW

**Review by rubn-g — APPROVED** [review]

**Review by Hoverbear — APPROVED** [review]

**Comment by Hoverbear** [comment]
I think the solution to the mapping you came up with is nice and simple. :)

**Comment by mloughran** [comment]
Very pleased to find this PR, and would not wish to delay it's merging :) (please say if it would be better to raise a separate PR).

Would you consider adding FATAL to the `remap_severity` function? My personal interest is allowing a clean mapping of Ruby's log levels (debug < info < warn < error < fatal), however the use of FATAL nomenclature instead of CRITIAL/ALERT/EMERGENCY seems fairly common, e.g.:

* Ruby https://ruby-doc.org/stdlib-2.4.0/libdoc/logger/rdoc/Logger.html
* Log4j https://logging.apache.org/log4j/2.x/manual/customloglevels.html
* Golang (nomenclature only, standard logger not leveled)

At the risk of opening a can of worms, I suspect it would also be useful to add TRACE. Google's stackdriver logs say `DEBUG: (100) Debug or trace information.`, and trace is used by e.g.:

* Rust https://docs.rs/log/0.4.0/log/enum.Level.html
* Log4j https://logging.apache.org/log4j/2.x/manual/customloglevels.html
* .NET https://docs.microsoft.com/en-us/aspnet/core/fundamentals/logging/?view=aspnetcore-3.1#log-level

How about mapping FATAL -> 800 & TRACE -> 100 ?

**Review by Hoverbear — APPROVED** [review]

**Comment by bruceg** [comment]
@binarylogic is what I did for documentation reasonable?

File: .meta/sinks/gcp_stackdriver_logs.toml.erb [hunk 1]
```diff
@@ -113,6 +113,25 @@ Exactly one of `billing_account_id`, `folder_id`, `organization_id`, or \
...
 
+[sinks.gcp_stackdriver_logs.options.severity_key]
+type = "string"
+required = false
+examples = ["severity"]
+description = """\
+The field of the log event from which to take the outgoing log's \
+`severity` field. The named field is removed from the log event if \
+present, and must be either an integer between 0 and 800 or a string \
+containing one of the severity level names (case is ignored) \
```
Comment (binarylogic) [inline .meta/sinks/gcp_stackdriver_logs.toml.erb:124]: Can you link to the severity levels here?

Comment (bruceg) [reply .meta/sinks/gcp_stackdriver_logs.toml.erb:124]: I believe I did just below that, but I'll add another link.

**Review by binarylogic — APPROVED** [review]

**Pull request merged**